### ISSUE TITLE
Form not sending emails

### ISSUE THREAD

**Issue by Stegosource** [CalderaWP/Caldera-Forms#676]
We set up a form and are using Mailgun as a 3rd party email delivery service. Mailgun is working properly when tested with Gravity Forms, but none of the Caldera notifications are even being recognized by Mailgun.

**Comment by Shelob9** [comment]
@Stegosource Is MailGun implemented as a wp_mail() dropin?  That should work. But, I would want to make sure it is getting to where we go to wp_mail() (in 1.3.5, that's here: https://github.com/CalderaWP/Caldera-Forms/blob/current-stable/classes/save.php#L372)

If you read up that class you will see there are a few places where is can return because mail data is invalid.  If you knew where, then we could say why it is invalid, as I'd bet that's the issue.

**Comment by Stegosource** [comment]
Yeah, I think it is supposed to hijack emails sent with wp_mail(). We're using their plugin (https://wordpress.org/plugins/mailgun/). I cant really say more than that right now. Strange that it works with Gravity Forms though. Does Caldera use wp_mail()?

**Comment by Shelob9** [comment]
@Stegosource So your issue motivated us to do #680 because we had no easy way to check if the data passed to wp_mail() was right. Now you do (upgrade to 1.3.6 and check the preview option in the mailer settings.) I would be very interested in a screenshot of your preview.

**Comment by Shelob9** [comment]
@Stegosource I'm testing with that plugin, and it seems to work for me. I'm super interested in knowing, but what you are passing to wp_mail -- you could catch those with `caldera_forms_do_mailer` filter -- and what the final args for wp_remote_post() in mailgun are -- you could catch them with `http_request_args` filter.

^^ That's  sensitive data you should probably put in a text file and send to me on FB or Slack BTW

This is one of those, probably user error, else massive fail on our part sort of issues.,,

**Comment by Shelob9** [comment]
@Stegosource  I'm closing this since we can't reproduce & have not heard back. Send me a FB message if you're still having issues.

**Comment by marcinjackowiak** [comment]
I ran into a similar issue my problem was with the $to variable.  Caldera uses an array and i was expecting a string wordpress specification does mention that either is allowed.  

I just looked at the Mailgun Plugin code and looks like maybe it has the same problem, i don't see any code  to implode the $to variable to an array.  Should look something like this:

if ( ! empty( $to ) && is_array( $to ) )
		$body['to'] = implode( ', ', $to );

I have not tested the mailgun plugin yet but my code is working after I added the proper handling for the $to array.
